src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -1921,6 +1921,34 @@ void Assembler::vcvtdq2pd(XMMRegister dst, XMMRegister src, int vector_len) {
   emit_int16((unsigned char)0xE6, (0xC0 | encode));
 }
 
+void Assembler::vcvtps2ph(XMMRegister dst, XMMRegister src, int imm8, int vector_len) {
+  assert(VM_Version::supports_evex() || VM_Version::supports_f16c(), "");
```

Comment:
Since you are accepting vector_len so adding a AVX512VL check will be more appropriate here.